common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -5455,6 +5455,22 @@ public static enum ConfVars {
         + ",hive.zookeeper.ssl.truststore.location"
         + ",hive.zookeeper.ssl.truststore.password",
         "Comma separated list of configuration options which should not be read by normal user like passwords"),
+    HIVE_CONF_PROPAGATE_EXEC_ENGINES("hive.conf.propagate.exec.engines",
+        "fs.s3.awsAccessKeyId"
```

Comment:
If configs are dumped in logs or available as files in the local directories of AM or tasks, this would be visible to end users which can be a serious risk.

Reply:
yes, if configs are dumped in logs then it's a security risk, but currently, we have no other way to support 'less secure' option
('more secure' uses hadoop credential provider)
I don't have strong opinions about this one, so I'm closing this as invalid, until we're not facing customer pressure to support it